go/multipooler/manager/manager.go

```diff
@@ -1108,13 +1070,302 @@ func (pm *MultiPoolerManager) ConfigureSynchronousReplication(ctx context.Contex
+	}
+	defer pm.unlock()
+
+	pm.logger.Info("UpdateSynchronousStandbyList called",
+		"operation", operation,
+		"standby_ids", standbyIDs,
+		"reload_config", reloadConfig,
+		"consensus_term", consensusTerm,
+		"force", force)
+
+	// Validate and update consensus term following consensus rules
+	if err := pm.validateAndUpdateTerm(ctx, consensusTerm, force); err != nil {
+		return err
+	}
+
+	// === Validation ===
+
+	// Validate operation
+	if operation == multipoolermanagerdata.StandbyUpdateOperation_STANDBY_UPDATE_OPERATION_UNSPECIFIED {
+		return mterrors.New(mtrpcpb.Code_INVALID_ARGUMENT, "operation must be specified")
+	}
+
+	// Validate standby IDs using the shared validation function
+	if err := validateStandbyIDs(standbyIDs); err != nil {
+		return err
+	}
+
+	// Check PRIMARY guardrails (pooler type and non-recovery mode)
+	if err := pm.checkPrimaryGuardrails(ctx); err != nil {
+		return err
+	}
+
+	// === Parse Current Configuration ===
+
+	// Get current synchronous_standby_names value
+	var currentValue string
+	err := pm.db.QueryRowContext(ctx, "SHOW synchronous_standby_names").Scan(&currentValue)
+	if err != nil {
+		pm.logger.Error("Failed to get current synchronous_standby_names", "error", err)
+		return mterrors.Wrap(err, "failed to get current synchronous_standby_names")
+	}
+
+	pm.logger.Info("Current synchronous_standby_names", "value", currentValue)
+
+	// Parse current configuration
+	cfg, err := parseSynchronousStandbyNames(currentValue)
+	if err != nil {
+		return err
+	}
+
+	// === Apply Operation ===
+
+	// Apply the requested operation
+	var updatedStandbys []*clustermetadatapb.ID
+	switch operation {
+	case multipoolermanagerdata.StandbyUpdateOperation_STANDBY_UPDATE_OPERATION_ADD:
+		updatedStandbys = applyAddOperation(cfg.StandbyIDs, standbyIDs)
+
+	case multipoolermanagerdata.StandbyUpdateOperation_STANDBY_UPDATE_OPERATION_REMOVE:
+		updatedStandbys = applyRemoveOperation(cfg.StandbyIDs, standbyIDs)
+
+	case multipoolermanagerdata.StandbyUpdateOperation_STANDBY_UPDATE_OPERATION_REPLACE:
+		updatedStandbys = applyReplaceOperation(standbyIDs)
+
+	default:
+		return mterrors.New(mtrpcpb.Code_INVALID_ARGUMENT,
+			fmt.Sprintf("unsupported operation: %s", operation.String()))
+	}
+
+	// Validate that the final list is not empty
+	if len(updatedStandbys) == 0 {
+		return mterrors.New(mtrpcpb.Code_INVALID_ARGUMENT,
+			"resulting standby list cannot be empty after operation")
+	}
+
+	// === Build and Apply New Configuration ===
+
+	// Build new synchronous_standby_names value
+	// Convert standby IDs to quoted application names
+	quotedNames := make([]string, len(updatedStandbys))
+	for i, id := range updatedStandbys {
+		appName := generateApplicationName(id)
+		quotedNames[i] = fmt.Sprintf(`"%s"`, appName)
+	}
+	standbyList := strings.Join(quotedNames, ", ")
+
+	// Convert method enum back to string
+	var methodStr string
+	switch cfg.Method {
+	case multipoolermanagerdata.SynchronousMethod_SYNCHRONOUS_METHOD_FIRST:
```

Comment:
Worth factoring out some shared code with setSynchronousStandbyNames(), like formatting the method + NumSync + standbyList into a string and/or some of the SQL statements?